openmeter/appstripe/adapter/stripe.go

```diff
@@ -0,0 +1,344 @@
+package appstripeadapter
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openmeterio/openmeter/openmeter/app"
+	appentity "github.com/openmeterio/openmeter/openmeter/app/entity"
+	appentitybase "github.com/openmeterio/openmeter/openmeter/app/entity/base"
+	"github.com/openmeterio/openmeter/openmeter/appstripe"
+	stripeclient "github.com/openmeterio/openmeter/openmeter/appstripe/client"
+	appstripeentity "github.com/openmeterio/openmeter/openmeter/appstripe/entity"
+	customerentity "github.com/openmeterio/openmeter/openmeter/customer/entity"
+	"github.com/openmeterio/openmeter/openmeter/ent/db"
+	entdb "github.com/openmeterio/openmeter/openmeter/ent/db"
+	appstripedb "github.com/openmeterio/openmeter/openmeter/ent/db/appstripe"
+	appstripecustomerdb "github.com/openmeterio/openmeter/openmeter/ent/db/appstripecustomer"
+	"github.com/openmeterio/openmeter/openmeter/secret"
+	secretentity "github.com/openmeterio/openmeter/openmeter/secret/entity"
+	"github.com/openmeterio/openmeter/pkg/framework/transaction"
+	"github.com/openmeterio/openmeter/pkg/models"
+)
+
+var _ appstripe.AppStripeAdapter = (*adapter)(nil)
+
+// CreateApp creates a new app
+func (a adapter) CreateStripeApp(ctx context.Context, input appstripeentity.CreateAppStripeInput) (appstripeentity.App, error) {
+	if err := input.Validate(); err != nil {
+		return appstripeentity.App{}, appstripe.ValidationError{
+			Err: fmt.Errorf("error create stripe app: %w", err),
+		}
+	}
+
+	return transaction.Run(ctx, a, func(ctx context.Context) (appstripeentity.App, error) {
+		// Create the base app
+		appBase, err := a.appService.CreateApp(ctx, appentity.CreateAppInput{
+			Namespace:   input.Namespace,
+			Name:        input.Name,
+			Description: input.Description,
+			Type:        appentitybase.AppTypeStripe,
+		})
+		if err != nil {
+			return appstripeentity.App{}, fmt.Errorf("failed to create app: %w", err)
+		}
+
+		// Create the stripe app in the database
+		appStripeCreateQuery := a.db.AppStripe.Create().
```

Comment:
Should we ensure that only one application can exist per stripe account?

Reply:
I think it's fine